Merge Conflict/Assets/Scripts/UiManager.cs

```diff
@@ -0,0 +1,186 @@
+    [SerializeField] private Button buttonSettings;
+    [SerializeField] private Button buttonLevel;
+    [SerializeField] private Button buttonUpgrade;
+    [SerializeField] private Button buttonElements;
+    [SerializeField] private Button buttonExitGame;
+
+    //all screens
+    [SerializeField] private Canvas PLAYFIELD;
+    [SerializeField] private Canvas UI_MANAGER;
+    [SerializeField] private Canvas MAINMENU;
+    [SerializeField] private Canvas SETTINGS;
+    [SerializeField] private Canvas LEVEL;
+    [SerializeField] private Canvas UPGRADE;
+    [SerializeField] private Canvas ELEMENTS;
+
+    //mapping buttons to the screens wich they should open
+    List<KeyValuePair<string, string>> buttonRelationScreen = new List<KeyValuePair<string, string>>
+    {
+        new KeyValuePair<string, string>("ButtonOpenMainmenu", "Mainmenu"),
+        new KeyValuePair<string, string>("ButtonCloseMainmenu", "CloseMenu"),
+        new KeyValuePair<string, string>("ButtonOpenSettings", "Settings"),
+        new KeyValuePair<string, string>("ButtonOpenLevel", "Level"),
+        new KeyValuePair<string, string>("ButtonOpenUpgrade", "Upgrade"),
+        new KeyValuePair<string, string>("ButtonOpenElements", "Elements"),
+    };
+
+    const Canvas NO_OPEN_SCREEN = null;
+    const string EXIT_GAME = "ButtonExitGame";
+    const bool OPENED = true;
+    const bool CLOSED = false;
+
+    private Canvas currentOpenedScreen;
+    private bool menuVisibility = CLOSED;
+
+
+    void Update()
+    {
+        Debug.Log("state: " + getMenuVisibility());
+    }
+
+    void Start()
+    {
+        //set default screen states 
+        UI_MANAGER.enabled = true;
+        MAINMENU.enabled = false;
+        SETTINGS.enabled = false;
+        LEVEL.enabled = false;
+        UPGRADE.enabled = false;
+        ELEMENTS.enabled = false;
+
+        //setup eventlisteners for all buttons
+        setupButtonListener(buttonOpenMainmenu);
+        setupButtonListener(buttonCloseMainmenu);
+        setupButtonListener(buttonSettings);
+        setupButtonListener(buttonLevel);
+        setupButtonListener(buttonUpgrade);
+        setupButtonListener(buttonElements);
+        setupButtonListener(buttonExitGame);
+    }
+
+    private void setupButtonListener(Button button)
+    {
+        if (button != null)
+        {
+            button.onClick.AddListener(() => handleButtonClick(button.name));
+        }
+        else
+        {
+            Debug.LogError($"Button with name {button} not found. Please check if the button exists and ist linkt to the script UiManager!");
+        }
+    }
+
+    private void handleButtonClick(string clickedButton)
+    {
+        if (clickedButton == EXIT_GAME)
+        {
+#if UNITY_EDITOR
+                UnityEditor.EditorApplication.isPlaying = false;
+#else
+            Application.Quit();
+#endif
+            return;
+        }
+
+        screenSwitcher(clickedButton);
+    }
+
+    private void screenSwitcher(string requestedScreen)
```

Comment:
rename to "switchScreen"

it is also unclear what the difference between "screen" and "Menu" is. If they are interchangable only use one of them

Reply:
ok, i will use menu for all.